### ISSUE TITLE
Improvements for the ClayPanel component

### ISSUE THREAD

**Issue by veroglez** [liferay/clay#5624]
Hi guys!

From the Echo team we are replacing all our Collapses with ClayPanel component, and doing this migration several questions have arisen:

When an item is clicked a focus appears, could we implement `c-inner` so that the focus only appears when the element is focused?

<img width="250" alt="Screen Shot 2023-07-06 at 2 32 24 PM" src="https://github.com/liferay/clay/assets/9701095/5758915d-8141-443a-9254-5dce25aa312d">

Also, we have also noticed that the styles are not the same as those defined in Lexicon for this component (see the following images). Is it possible to implement them?

_Current style_
<img width="288" alt="Screen Shot 2023-07-06 at 2 33 28 PM" src="https://github.com/liferay/clay/assets/9701095/b08e0523-97aa-48e4-8e4d-e29a657afacf">

_Styles defined by Lexicon_
![image](https://github.com/liferay/clay/assets/9701095/5e84f21d-183b-451c-86d9-0d04ef2fcb40)

Thank you very much in advance!

**Comment by matuzalemsteles** [comment]
> When an item is clicked a focus appears, could we implement c-inner so that the focus only appears when the element is focused?

Yeah, I think it's possible! not sure if this conflicts with the new accessibility menu rule that this will be activated by the user with focus visible. Thoughts @ethib137 @marcoscv-work 

> Also, we have also noticed that the styles are not the same as those defined in Lexicon for this component (see the following images). Is it possible to implement them?

cc @pat270

**Comment by marcoscv-work** [comment]
Yeah, no problem on removing focus for mouse interaction. that's something Lexicon always wanted and we can apply as much as possible.
WCAG 2.2 will be more specific about this, but it also specify it applies for keyboard interaction / screen reader (they have their own focus):
https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance-minimum.html

**Comment by ethib137** [comment]
I think the questions we need to be asking are the following:
- What should the default behavior be?
    - ie: focus-visible-only (only display focus for keyboard interaction) or focus-visible (display focus for both keyboard and mouse interactions)
- What should the accessibility menu setting be ([Other settings for Reference](https://liferay.atlassian.net/wiki/spaces/PEDS/pages/2228781375/Accessibility+Menu+and+Shortcut+modal+to+allow+implementing+accessibility+solutions+faster+and+easier+and+with+more+awareness+for+end+users#List-of-initial-configurations-to-add))?
    - Label?
    - Description?
    - Default Value?
    - CSS Class?

From the discussion thus far it sounds like we are proposing the following:
- What should the default behavior be?
    - focus-visible-only (only display focus for keyboard interaction) 
- What should the accessibility menu setting be?
    - Label: Focus Indicator
    - Description: Show the focus indicator when either the mouse or keyboard is used.
    - Default Value: False
    - CSS Class: "c-prefers-focus-indicator"
    
[Related Issue](https://github.com/liferay/clay/pull/5549)

**Comment by marcoscv-work** [comment]
I think for this question @veroglez is speaking about, we need to use focus-visible-only (only for keyboard). we already had this DXP component working in this way in portal and they are trying to use the clay component so for this case I think is clear.

From a design perspective we can use always focus-visible-only. since our interactive components should have always  `hover` state.


Speaking about accessibility menu:

We have components that are showing the focus sometimes with mouse interactions. Lexicon dint wanted this but it was a initial technical limitation, it's something understandable, we all know that adding c-inner for all clay components/DXP components/taglibs/custom markup, etc.. would be a complex task, for that reason we were looking for a way to completely deactivate this effect.

If now we can do it focus-visible-only (only for keyboard) globally without the need of use c-inner, I would go for that as a default value in DXP and I would redefine the accessibility menu option, we could change it of just remove it.

**Comment by matuzalemsteles** [comment]
> If now we can do it focus-visible-only (only for keyboard) globally without the need of use c-inner, I would go for that as a default value in DXP and I would redefine the accessibility menu option, we could change it of just remove it.

I think this comes back into the discussion about this in the implementation of PR #5006.

**Comment by matuzalemsteles** [comment]
LPS Ticket https://liferay.atlassian.net/browse/LPS-192228
